assets/src/stories-editor/helpers/isBlockAllowedOnPage.js

```diff
@@ -0,0 +1,24 @@
+/**
+ * WordPress dependencies
+ */
+
+import { select } from '@wordpress/data';
+
+const {
+	canInsertBlockType,
+	getBlockListSettings,
+} = select( 'core/block-editor' );
+
+/**
+ * Is the given block allowed on the given page?
+ *
+ * @param {Object}  name The name of the block to test.
+ * @param {string}  pageId Page ID.
+ * @return {boolean} Returns true if the element is allowed on the page, false otherwise.
+ */
+export default ( name, pageId ) => {
```

Comment:
Should this be a custom hook similar to `useMoveBlockToPage`, as you are using a state data?

Reply:
Yes, you're right. It should. It would be beneficial to update all helpers using `select` or `dispatch` to be hooks in order for them to properly update components that use them. I will update this one.

Reply:
Argh, it can't easily be as it's used in a class component, `InserterMenu`. And that's a pretty complex component I don't feel like rewriting right now.

I will split it into two copies and deprecate the non-hook one.

I don't know if there's a better transition method, I'm open for suggestions.

Reply:
There are 15 selects and 11 dispatches in the helper functions. These should all be convert to custom hooks. We should follow up with another ticket.

Reply:
> Argh, it can't easily be as it's used in a class component, InserterMenu. And that's a pretty complex component I don't feel like rewriting right now.

I think this is related to what you are saying. https://github.com/ampproject/amp-wp/pull/3412 . Might be useful.

Reply:
Not all of them are used in actual components though FWIW

Reply:
Attempted fixed in e963844.